udaykalra/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,187 @@
+          overlappingEndMinute, overlappedTimeRangeEndMinute - overlappingEndMinute);
+      replacementRanges.add(replaceTime);
+    }
+
+    // Case 2&3: Trim end of overlappedTimeRange.
+    else if ((overlapType == OVERLAP_WITH_SAME_END) || (overlapType == OVERLAPS_END)) {
+      TimeRange replaceTime = TimeRange.fromStartDuration(
+          overlappedTimeRangeStartMinute, overlappingStartMinute - overlappedTimeRangeStartMinute);
+      replacementRanges.add(replaceTime);
+    }
+
+    // Case 5: Chop overlappingTimeRange out of overlappedTimeRange.
+    else if (overlapType == OVERLAP_CONTAINS) {
+      TimeRange replaceTimeA = TimeRange.fromStartDuration(
+          overlappedTimeRangeStartMinute, overlappingStartMinute - overlappedTimeRangeStartMinute);
+      TimeRange replaceTimeB = TimeRange.fromStartDuration(
+          overlappingEndMinute, overlappedTimeRangeEndMinute - overlappingEndMinute);
+      replacementRanges.add(replaceTimeA);
+      replacementRanges.add(replaceTimeB);
+    } else {
+      replacementRanges.add(overlappedTimeRange);
+    }
+    return replacementRanges;
+  }
+
+  /**
+   * Returns integer representing overlap case type:
+   *  1. OVERLAP_WITH_SAME_START - TimeRanges start at same time and overlappingTimeRange
+   *     ends sooner.
+   *  2. OVERLAP_WITH_SAME_END - overlappingTimeRange starts during timeRange and
+   *     both end together.
+   *  3. OVERLAPS_END - overlappingTimeRange starts during timeRange and ends after.
+   *  4. OVERLAPS_START - overlappingTimeRange starts before timeRange and ends
+   *     during.
+   *  5. OVERLAP_CONTAINS - overlappingTimeRange is completely overlapped by
+   *     timeRange.
+   *  0. Does not overlap
+   */
+  private int classifyOverlap(TimeRange overlappingTimeRange, TimeRange timeRange) {
+    // Obtain timeRange and overlappingTimeRange start/end data..
+    int timeRangeStartMinute = timeRange.start();
+    int timeRangeEndMinute = timeRange.end();
+
+    int overlappingStartMinute = overlappingTimeRange.start();
+    int overlappingEndMinute = overlappingTimeRange.end();
+
+    // Define Overlap Cases
+    if ((timeRangeStartMinute == overlappingStartMinute)
+        && (timeRangeEndMinute > overlappingEndMinute)) {
+      return OVERLAP_WITH_SAME_START;
+    }
+
+    else if ((timeRangeEndMinute == overlappingEndMinute)
+        && (timeRangeStartMinute < overlappingStartMinute)) {
+      return OVERLAP_WITH_SAME_END;
+    }
+
+    else if ((timeRangeStartMinute < overlappingStartMinute)
+        && (timeRangeEndMinute < overlappingEndMinute)) {
+      return OVERLAPS_END;
+    }
+
+    else if ((timeRangeStartMinute > overlappingStartMinute)
+        && (timeRangeEndMinute > overlappingEndMinute)) {
+      return OVERLAPS_START;
+    }
+
+    else if ((timeRangeStartMinute < overlappingStartMinute)
+        && (timeRangeEndMinute > overlappingEndMinute)) {
+      return OVERLAP_CONTAINS;
+    } else {
+      return 0;
```

Comment:
Instead of 0, could you return a NO_OVERLAP constant? That would make it more obvious for the user of this method to know what value to check for.